datafusion/sql/src/planner.rs

```diff
@@ -103,6 +103,31 @@ pub struct ParserOptions {
     parse_float_as_decimal: bool,
 }
 
+#[derive(Debug, Clone)]
+/// Struct to store Common Table Expression provided with WITH clause and Parameter Data Types provided with PREPARE statement
+pub struct Ctes {
+    /// Data type provided with prepare statement
+    pub prepare_param_data_types: Vec<DataType>,
+    /// Map of CTE name to logical plan of the WITH clause
+    pub ctes: HashMap<String, LogicalPlan>,
+}
+
+impl Default for Ctes {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Ctes {
+    /// Create a new Ctes
+    pub fn new() -> Self {
+        Self {
+            prepare_param_data_types: vec![],
```

Comment:
This is always empty in this PR. After this is merged, I will use and set it in PR #4490